README.md

```diff
@@ -635,6 +635,71 @@ In the event that something happened while trying to save or retrieve the creden
 - Device's Lock Screen security settings have changed (e.g. the PIN code was changed). Even when `hasCredentials` returns true, the encryption keys will be deemed invalid and until `saveCredentials` is called again it won't be possible to decrypt any previously existing content, since they keys used back then are not the same as the new ones.
 - Device is not compatible with some of the algorithms required by the `SecureCredentialsManager` class. This is considered a catastrophic event and might happen when the OEM has modified the Android ROM removing some of the officially included algorithms. Nevertheless, it can be checked in the exception instance itself by calling `isDeviceIncompatible`. By doing so you can decide the fallback for storing the credentials, such as using the regular `CredentialsManager`.
 
+## Networking client customization
+
+This library provides the ability to customize the behavior of the networking client for common configurations, as well the ability to define and use your own networking client implementation.
+
+The API clients and `WebAuthProvider` can be configured with a `NetworkingClient`, which will be used when making requests. You can configure the default client with custom timeout values, any headers that should be sent on all requests, and whether to log request/response info (for non-production debugging purposes only). For more advanced configuration, you can provide your own implementation of `NetworkingClient`
+
+### Timeout configuration
+
+```kotlin
+val account = Auth0("{YOUR_CLIENT_ID}", "{YOUR_DOMAIN}")
+
+val netClient = DefaultClient(
+    connectTimeout = 30,
+    readTimeout = 30
+)
+
+val authAPI = AuthenticationAPIClient(account, netClient)
+```
+
+### Logging configuration
+
+```kotlin
+val account = Auth0("{YOUR_CLIENT_ID}", "{YOUR_DOMAIN}")
+
+val netClient = DefaultClient(
+    enableLogging = true
+)
+
+val authAPI = AuthenticationAPIClient(account, netClient)
+```
+
+### Set additional headers for all requests
+
+```kotlin
+val account = Auth0("{YOUR_CLIENT_ID}", "{YOUR_DOMAIN}")
+
+val netClient = DefaultClient(
+    defaultHeaders = mapOf("{HEADER-NAME}" to "{HEADER-VALUE}")
+)
+
+val authAPI = AuthenticationAPIClient(account, netClient)
+```
+
+### Advanced configuration
+
+For more advanced configuration of the networking client, you can provide a custom implementation of `NetworkingClient`. This may be useful when you wish to reuse your own networking client, configure a proxy, etc.
+
+```kotlin
+class CustomNetClient : NetworkingClient {
+    override fun load(url: String, options: RequestOptions): ServerResponse {
+        val response = // create and execute the request
```

Comment:
".. with the given options" ? or something pointing out that the details that might be required to build it are passed through the parameters

Reply:
Added that info to the comment in a new commit.